adr/ADR-0009-project-workflow.md

```diff
@@ -0,0 +1,623 @@
+       GH->>K8s: 🤖 Deploy all components
+       GH->>Tests: 🤖 Run E2E tests
+       Tests->>GH: 🤖 Report results
+       GH->>PR: 🤖 Update status
+       Dev->>PR: 👤 Review results and merge if tests pass
+   ```
+
+   **Concrete Example: Orchestrator PR Testing**
+
+   For a PR on the guardrails orchestrator component:
+
+    1. 👤 Developer creates PR #42 to the upstream orchestrator repository
+    2. 🤖 GitHub Actions triggers the E2E workflow automatically
+    3. 🤖 Workflow builds `quay.io/trustyai/fms-guardrails-orchestrator:PR-42`
+    4. 🤖 Workflow checks out the operator repository
+    5. 🤖 Workflow updates only the orchestrator image in `params.env`:
+       ```diff
+       # params.env
+       TRUSTYAI_SERVICE_IMAGE=quay.io/trustyai/service:latest
+       - TRUSTYAI_ORCHESTRATOR_IMAGE=quay.io/trustyai/fms-guardrails-orchestrator:latest
+       + TRUSTYAI_ORCHESTRATOR_IMAGE=quay.io/trustyai/fms-guardrails-orchestrator:PR-42
+       TRUSTYAI_DETECTOR_IMAGE=quay.io/trustyai/detector:latest
+       TRUSTYAI_RUNTIME_IMAGE=quay.io/trustyai/runtime:latest
+       ```
+    6. 🤖 Workflow deploys the operator and all components to test cluster
+    7. 🤖 Workflow runs E2E tests against the deployment
+    8. 🤖 Test results are reported back to the PR
+    9. 👤 Developers review the results and merge if tests pass
+
+2. **Daily Sync from Upstream to Midstream Main**:
+
+   A GitHub Actions workflow that runs on a schedule to sync changes from **upstream:main** to **midstream:main**. The workflow:
+
+    - Runs daily at a pre-determined time
+    - Checks out the midstream repository
+    - Adds the upstream repository as a remote
+    - Fetches from upstream
+    - Merges upstream/main into midstream/main
+    - Runs basic validation tests
+    - Pushes to midstream main
+    - Notifies the team if sync fails (_e.g._, via Slack)
+
+   ```mermaid
+   sequenceDiagram
+       participant U as Upstream:main
+       participant GH as GitHub Actions 🤖
+       participant M as Midstream:main
+       participant N as Notifications
+       participant Team as Team Members 👤
+
+       Note over GH: 🤖 Daily trigger
+       GH->>M: 🤖 Checkout midstream
+       GH->>GH: 🤖 Add upstream remote
+       GH->>U: 🤖 Fetch changes
+       GH->>GH: 🤖 Merge upstream/main
+       GH->>GH: 🤖 Run validation tests
+       GH->>M: 🤖 Push changes
+
+       alt Sync Failed
+           GH->>N: 🤖 Create issue for team
+           N->>Team: 👤 Team reviews sync failure
+           Team->>M: 👤 Manual intervention if needed
+       end
+   ```
+
+3. **ODH Release Creation Workflow**:
+
+   A manually triggered GitHub Actions workflow that creates ODH release branches from the incubation branch. The workflow:
+
+    - Accepts the release version input (e.g., 2.20, 2.21)
+    - Checks out the incubation branch
+    - Creates a new branch named `odh/[version]`
+    - Tags the branch with `v[version]-odh`
+    - Triggers image builds for the release
```

Comment:
here will be some extra steps (set up tekton pipeline etc)  if use konflux

Reply:
Made a comment on Konflux setup (although Konflux pipeline details are out of scope for this proposal)